### PR TITLE
Fix exponential slowdown in XML normalizer

### PR DESCRIPTION
**Pull request by millerdev**
http://manage.dimagi.com/default.asp?236305

Review commits individually.

@emord @orangejenny

### CODE REVIEW

File: tests/xml.js [hunk 1]
```diff
@@ -150,6 +152,18 @@ define([
             eq('<output attr=value />', '<output attr="value" />', false);
         });
 
+        it("should fail on escape unquoted attribute value with <", function () {
+            chai.expect(function () {
+                eq('<output value=2<3 />', '<output value="2&lt;3" />', false);
```
Comment (emord) [inline tests/xml.js]: What would `normalize` or `humanize` return? or is this just saying that they won't match each other?

Comment (millerdev) [inline tests/xml.js]: I don't know if that input is even valid XML for any parser, but it seems like it might be possible to interpret the first string as the second. However, it will not work with our `normalize` (or `humanize` either) because `fixGTBug` can't handle that input. I added these known edge case failures as documentation of things we know will not work.

Comment (millerdev) [inline tests/xml.js]: The result of `xml.normalize('<output value=2<3 />')` on phantomjs is

`<output value="2&lt;3"><!--3--></output>`

which is very strange indeed. It's even semantically equivalent XML, which is nice.

File: tests/xml.js [hunk 2]
```diff
@@ -150,6 +152,18 @@ define([
...
+            }).to.throw(Error);
+        });
+
+        it("should fail on escape attribute value with />", function () {
+            chai.expect(function () {
+                eq('<output value="../>.." />', '<output value="..&gt;.." />', false);
```
Comment (millerdev) [inline tests/xml.js]: This input is really botched by `fixGTBug`:

`<output value="..&gt;&lt;/output&gt;.." />`

File: tests/xml.js (hunk 1 above)
Comment (emord) [inline tests/xml.js]: Ah. I didn't realize this was an already known error. Should there be a comment on why/what we expect this to fail with an error? It looks like eq has two asserts that could error

Comment (millerdev) [inline tests/xml.js]: https://github.com/dimagi/Vellum/pull/748

**Comment by phillipm** [comment]
Woohoo! Excited to see this fixed!
